### ISSUE TITLE
Scrolling reads all inputs rather than just most recent

### ISSUE THREAD

**Issue by that1communist** [greshake/i3status-rust#101]
If you scroll on a trackpad on the volume icon, you'll notice it won't stop when you stop scrolling, rather it will go through all the inputs you gave it, slowly.

I don't know best how to describe this issue, sorry if the title is poor, I can resubmit.

**Comment by greshake** [comment]
Can't reproduce here, more details please :)

**Comment by that1communist** [comment]
https://photos.app.goo.gl/V0NuUt5giVQ7ncYn1

Here's a video.

even if i move the cursor away after i've done the scrolling, it will continue to change the volume.

**Comment by greshake** [comment]
Okay. This is what we need:
1. OS/ Versions of software involved (i3, i3status-rust, i3bar, do you use a compositor?) 
2. Configuration files for i3, i3status-rust

What you can try to help debug the issue:
- Does the problem occur when using a regular mouse?
- Add another block that reacts to clicks (e.g. music or temp) and see if they react instantly to clicking or are delayed

If no lead turns up, I'll try to sprinkle a few debugging statements and we'll do a logging run :)

**Comment by that1communist** [comment]
1. OS/ Versions of software involved (i3, i3status-rust, i3bar, do you use a compositor?)

The issue happens without a compositor.

```
i3 version 4.14 (2017-09-04)
i3bar version 4.14 (2017-09-04)
i3status-rs 0.9
```


Configuration files for i3, i3status-rust

This is the bar section of my i3 config
```
bar {
  bindsym button1 nop
  status_command /usr/bin/i3status-rs ~/.i3/config.toml
  font pango:Noto Mono, fontawesome 7
  strip_workspace_numbers yes
	colors {
		background		#000000
		statusline		#ffffff
		separator		#383838
		focused_workspace	#ff0000 #440000 #ffffff
		active_workspace	#ff0000 #440000 #383838
		inactive_workspace	#383838 #383838 #ffffff
		urgent_workspace	#d32f2f #d32f2f #000000
    	}
}
```
This is my i3status-rust config
[theme]
name = "plain"
[theme.overrides]
idle_bg = "#000000"
idle_fg = "#FFFFFF"
info_bg = "#000000"
info_fg = "#FFFFFF"
good_bg = "#000000"
good_fg = "#FFFFFF"
warning_bg = "#000000"
warning_fg = "#FFFFFF"
critical_bg = "#000000"
critical_fg = "#FFFFFF"
separator = "20"
separator_bg = "#000000"
separator_fg = "#000000"

```
[icons]
name = "awesome"

[[block]]
block = "sound"

[[block]]
block = "battery"

[[block]]
block = "time"
interval = 60
format = "%a∣%d-%b∣%H:%M"

```

What you can try to help debug the issue:

I have no idea, I know nothing about rust programming.

Does the problem occur when using a regular mouse?

Yes, if you scroll fast enough you will see it on a regular mouse as well, you do have to scroll extremely fast though.

Add another block that reacts to clicks (e.g. music or temp) and see if they react instantly to clicking or are delayed

About to do this and return to editing this post, but I suspect it will.

**Comment by greshake** [comment]
4. To Clicking, not scrolling

To your last point: That's very difficult to achieve if not impossible with i3bar. The spacing largely depends on the font, and most fonts don't use constant size characters. But, you can simply choose a monospace font to fix the issue. In that case, there is only expansion when reachind 100% volume- that should be bearable, otherwise we always waste one character in width for a 0 or space.

**Comment by that1communist** [comment]
Okay but my i3blocks setup already does that. Icon and text don't move at all. 

As for the actual issue here, I think this issue is a performance one from a delay in the changing of volume.

**Comment by greshake** [comment]
See if https://github.com/greshake/i3status-rust/pull/108 fixed it

**Comment by that1communist** [comment]
It didn't fix the scrolling behavior, but it did definitely make it more performant.


It still treats the trackpad like a trackball.

**Comment by that1communist** [comment]
I have realized that it might be because there is a delay between events on i3blocks, meaning there's a maximum scroll speed.

This could be wrong, but its another possibility.

**Comment by greshake** [comment]
No, this definitely works totally fine for me, even with fast interaction.

**Comment by that1communist** [comment]
Your computer might just be too fast. It is way harder to reproduce this bug on my desktop than on my craptop.

**Comment by greshake** [comment]
Well, that may be true, but the requirements of i3status-rust and i3bar are comically low compared to other software today. Take Spotify Desktop, an application using Electron (afaik)- Meaning you have a full blown Chrome instance running a web version of Spotify. That is eating up to 15% CPU idling on my i7. The point is, if todays 'normal' software works on your craptop, mine should work on a craptop one third as good.

**Comment by that1communist** [comment]
Then I have no idea what is causing this issue. I have access to a much more powerful laptop this weekend and I'll test it on that then, but even then, this should be more performant than i3blocks, so I don't know.

**Comment by ammgws** [comment]
@Kommynct Did you ever end up testing on your other PC?
